File: library1/src/main/java/com/gcc/library1/model/Book.java
package com.gcc.library1.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "books")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String isbn;

    private String title;

    private String author;

    @Column(length = 2000)
    private String description;

    // 图书封面图片（Base64 编码存储）
    @Column(columnDefinition = "TEXT")
    private String cover;

    // 图书分类
    private String category;

    // 库存数量
    @Column(nullable = false)
    private int count = 0;

    // 被借数量
    @Column(nullable = false)
    private int borrowCount = 0;

}

File: library1/html/flutter/lib/features/books/domain/book.dart
// 图书实体
class Book {
  final int id;
  final String title;
  final String author;
  final String isbn;
  final String publisher;
  final String category;
  final String? cover;
  final int count;
  final int borrowCount;

  const Book({
    required this.id,
    required this.title,
    required this.author,
    required this.isbn,
    required this.publisher,
    required this.category,
    this.cover,
    required this.count,
    required this.borrowCount,
  });

  // 从 JSON 反序列化
  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      author: json['author'] ?? '',
      isbn: json['isbn'] ?? '',
      publisher: json['publisher'] ?? '',
      category: json['category'] ?? '',
      cover: json['cover'],
      count: json['count'] ?? 0,
      borrowCount: json['borrowCount'] ?? 0,
    );
  }

  // 序列化为 JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'author': author,
      'isbn': isbn,
      'publisher': publisher,
      'category': category,
      'cover': cover,
      'count': count,
      'borrowCount': borrowCount,
    };
  }

  // 是否可借阅
  bool get isAvailable => count > 0;
}
